tests/dialects/test_print.py

```diff
@@ -0,0 +1,31 @@
+from xdsl.dialects import print, test, builtin
+from xdsl.transforms import print_to_println
+import pytest
+
+
+@pytest.mark.parametrize(
+    "given,expected",
+    (
+        ("test", "test"),
+        ("@hello  world", "hello_world"),
+        ("something.with.dots", "something_with_dots"),
+        ("this is 💩", "this_is"),
```

Comment:
lol

Reply:
Emojis will be kept around in the actual string, this builds the global symbol name (and only part of it)

Reply:
Can you add "(" and ")" as test cases to see what happens? Preferably end-to-end, like what would this do:
``` python
print("(")
print("a")
print(")")
```
Would it be the hash that resolves the difference between the two?

Reply:
Just a sidecomment (that does not even help here, AFAIU):

A classic, hacky trick in c++ to get unique numbers is to malloc a single byte and use the pointer to it as the unique number. Maybe you can do something similar here?

Reply:
I added a pytest test for this, I feel it's useful to test this behavior!

Reply:
Why not add them here

Reply:
If I added them here, the test would have to check the exact hashes, and would therefore test an implementation detail.

Instead, I want to test the contract. So that `key(x) == key(x)` and `key(x) != key(y)`, even for non-legal characters.

Reply:
I can do that in pytest much easier then I can in filecheck